ROADMAP.md

```diff
@@ -0,0 +1,359 @@
+
+## Vision
+
+_This section is still a very high-level draft_
+
+Testground As A Service embodies our long-term vision.
+
+- A single, scalable, installation that one or more organizations can use in all their projects,
+- The ability to experiment with large-scale networks and simplify the integration testing of "any" network-related code,
+- The ability to track the impact of a change in terms of stability & performance across multiple projects,
+  - An example: having the ability to run IPFS benchmarks and simulations with different combinations of libraries. This would help us measure regression and improvements as soon as they occur in upstream dependencies.
+
+Products with similar ideas but specialized in different areas:
+
+- database: [CockroachDB performance tracker](https://cockroachdb.github.io/pebble/?max=local),
+- browser: [Webkit Performance Dashboard](https://perf.webkit.org/v3/)
+
+Research and Templates
+
+- [Mockup for the Testground Dashboard](https://github.com/testground/testground/issues/133#issuecomment-553538029)
+
+## Our Focus for 2022-2023
+
+We focus on the following:
+
+1. Reliability: above all, Testground should be trusted by its users;
+2. Sustainability: implementing the processes & tools we need to maintain Testground in the medium and long term;
+3. Usefulness: solving needs that have been requested explicitly by projects.
+
+We want to ensure Testground is valuable and stable before we grow its feature set.
+
+## Milestones
+
+### 1. Bootstrap libp2p's interoperability testing story
+
+- Delivery: Q4 2022
+- Theme: usefulness
+- Effort: approx. 6 months
+
+**Why:** Testground provides an excellent foundation for distributed testing. Supporting the libp2p team with interoperability tests generates long-term value outside of Testground. We can move faster, generate interest, and create measurable improvements by focusing on a use-case.
+
+**Deliverables:**
+
+- Tooling to support interoperability testing (mixed-builders, composition templating),
+- Stability measurements & fixes to reach the libp2p team's expectations,
+- A fully working example (ping test) used in go-libp2p and rust-libp2p CIs,
+- An interoperability Dashboard that shows how implementations and versions are tested.
+
+### 2. Refresh Testground's EKS support
+
+- Delivery: Q4 2022
+- Theme: usefulness
+- Effort: approx. 6 months
+
+**Why:** Testground can simulate small networks in CI, but it covers more use cases when it lives in a larger cluster. When we run Testgroun in Kubernetes, we can support whole organizations through the Testground As A Service product.
+
+Using a managed service (Amazon's Elastic Kubernetes Service) means our maintenance costs are lower, and the team can focus on improvements.
+
+**Deliverables:**
+
+- An EKS installation script,
+  - Extra care is taken on Network infrastructure (CNIs).
+- A (fixed) Kubernetes runner that runs on Amazon's EKS,
```

Comment:
Does this one entail actually having a running cluster PL projects can submit jobs to? I think maybe it deserves a separate deliverable.

Reply:
I see, it's another milestone!